### DISCUSSION TITLE
Is there a simple template for a debugger which starts an external .exe application?

### DISCUSSION THREAD

**Discussion by Yazwh0** [microsoft/vscode-discussions#474] [Extension Development QnA] [answered]
I'm trying to use the mono debugger `launch.json` as a template, but try as I might VSCode simply ignores or silently fails the configuration for the `debuggers` contributes section. (Languages and grammar are working fine.)

Given I can't find a error message anywhere as to why its not working, I was hoping there was a decent template somewhere which I could start from?

Specifically if everything is handled in the external application, so I shouldn't need a `extension.ts`, so how VSCode can be configured to start the application.

All the examples seem to be about debugging something in TS\Javascript, which makes it really hard to understand where the boundaries are, eg, which part is extension, which part is the debug adapter.

**Comment by isidorn** [comment, marked as answer]
These docs should help https://code.visualstudio.com/api/extension-guides/debugger-extension
